Mage.Sets/src/mage/cards/y/YarusRoarOfTheOldGods.java

```diff
@@ -0,0 +1,118 @@
+        this.subtype.add(SubType.DRUID);
+        this.power = new MageInt(4);
+        this.toughness = new MageInt(4);
+
+        // Other creatures you control have haste.
+        this.addAbility(new SimpleStaticAbility(new GainAbilityControlledEffect(
+                HasteAbility.getInstance(), Duration.WhileOnBattlefield,
+                StaticFilters.FILTER_PERMANENT_CREATURES, true)));
+        // Whenever one or more face-down creatures you control deal combat damage to a player, draw a card.
+        this.addAbility(new DealCombatDamageControlledTriggeredAbility(
+                new DrawCardSourceControllerEffect(1), filter)
+                .setTriggerPhrase("Whenever one or more face-down creatures you control deal combat damage to a player, "));
+        // Whenever a face-down creature you control dies, return it to the battlefield face down under its owner's control if it's a permanent card, then turn it face up.
+        this.addAbility(new DiesCreatureTriggeredAbility(
+                new YarusRoarOfTheOldGodsEffect(), false, filter, true));
+    }
+
+    private YarusRoarOfTheOldGods(final YarusRoarOfTheOldGods card) {
+        super(card);
+    }
+
+    @Override
+    public YarusRoarOfTheOldGods copy() {
+        return new YarusRoarOfTheOldGods(this);
+    }
+}
+
+class YarusRoarOfTheOldGodsEffect extends OneShotEffect {
+
+    YarusRoarOfTheOldGodsEffect() {
+        super(Outcome.PutCardInPlay);
+        this.staticText = "return it to the battlefield face down under its owner's control " +
+                "if it's a permanent card, then turn it face up";
+    }
+
+    private YarusRoarOfTheOldGodsEffect(final YarusRoarOfTheOldGodsEffect effect) {
+        super(effect);
+    }
+
+    @Override
+    public YarusRoarOfTheOldGodsEffect copy() {
+        return new YarusRoarOfTheOldGodsEffect(this);
+    }
+
+    @Override
+    public boolean apply(Game game, Ability source) {
+        Player controller = game.getPlayer(source.getControllerId());
+        Card card = game.getCard(getTargetPointer().getFirst(game, source));
+        if (controller == null || card == null || !card.isPermanent(game)) {
+            return false;
+        }
+
+        Ability newSource = source.copy();
+        newSource.setWorksFaceDown(true);
+        MageObjectReference mor = new MageObjectReference(card.getId(),
+                card.getZoneChangeCounter(game) + 1, game);
+        game.addEffect(new BecomesFaceDownCreatureEffect(
+                null, mor, Duration.Custom,
+                BecomesFaceDownCreatureEffect.FaceDownType.MANUAL), newSource);
+        controller.moveCards(card, Zone.BATTLEFIELD, source, game, false, true, true, null);
+
+        // Need to process the move so that any ETB triggers on the card don't fire
+        game.getState().processAction(game);
+
+        Permanent permanent = game.getPermanent(card.getId());
+        return permanent != null && permanent.turnFaceUp(source, game, source.getControllerId());
```

Comment:
Must return true all the time here. True/false logic in most apply’s use cases: if something changed in game state then return true.